pkg/controller/appinst_api.go

```diff
@@ -409,6 +409,37 @@ func removeAppInstFromRefs(appInstKey *edgeproto.AppInstKey, appInstRefs *[]edge
 	return refsChanged
 }
 
+// getAppInstURI get the Uri for the application instance.
+// Takes into account how the app is deployed and features of a given cloudlet
+func getAppInstURI(ctx context.Context, appInst *edgeproto.AppInst, app *edgeproto.App, clusterInst *edgeproto.ClusterInst, cloudlet *edgeproto.Cloudlet, cloudletFeatures *edgeproto.PlatformFeatures) string {
+	// Internal apps, or apps that don't have ports exposed don't need uri
+	ports, _ := edgeproto.ParseAppPorts(app.AccessPorts)
+	if app.InternalPorts || len(ports) == 0 {
+		return ""
+	}
+
+	// Default to dedicated IP
+	uri := getAppInstFQDN(appInst, cloudlet)
+
+	// uri is specific to appinst if it has dedicated IP
+	if appInst.DedicatedIp {
+		return uri
+	}
```

Comment:
You can just change line 426 to
```
return getAppInstFQDN(appInst, cloudlet)
```
and avoid computing uri on line 422 when it's not needed.

Reply:
I updated the logic, and got rid of the `uri` local var